src/Avalonia.Base/PriorityValue.cs

```diff
@@ -234,52 +235,56 @@ private PriorityLevel GetLevel(int priority)
+            delayedSetter.SetAndNotify(this, (update, notify) =>
             {
-                var old = _value;
+                var val = update.value;
+                var notification = val as BindingNotification;
+                object castValue;
 
-                if (_validate != null && castValue != AvaloniaProperty.UnsetValue)
+                if (notification != null)
                 {
-                    castValue = _validate(castValue);
+                    val = (notification.HasValue) ? notification.Value : null;
                 }
 
-                ValuePriority = priority;
-                _value = castValue;
-
-                if (notification?.HasValue == true)
+                if (TypeUtilities.TryConvertImplicit(_valueType, val, out castValue))
                 {
-                    notification.SetValue(castValue);
-                }
+                    var old = _value;
 
-                if (notification == null || notification.HasValue)
-                {
-                    Owner?.Changed(this, old, _value);
-                }
+                    if (_validate != null && castValue != AvaloniaProperty.UnsetValue)
+                    {
+                        castValue = _validate(castValue);
+                    }
 
-                if (notification != null)
+                    ValuePriority = priority;
+                    _value = castValue;
+
+                    if (notification?.HasValue == true)
+                    {
+                        notification.SetValue(castValue);
+                    }
+
+                    if (notification == null || notification.HasValue)
+                    {
+                        notify(() => Owner?.Changed(this, old, _value));
+                    }
+
+                    if (notification != null)
+                    {
+                        Owner?.BindingNotificationReceived(this, notification);
+                    }
+                }
+                else
                 {
-                    Owner?.BindingNotificationReceived(this, notification);
+                    Logger.Error(
+                        LogArea.Binding,
+                        Owner,
+                        "Binding produced invalid value for {$Property} ({$PropertyType}): {$Value} ({$ValueType})",
+                        Property.Name,
+                        _valueType,
+                        val,
+                        val?.GetType());
                 }
-            }
-            else
-            {
-                Logger.Error(
-                    LogArea.Binding, 
-                    Owner,
-                    "Binding produced invalid value for {$Property} ({$PropertyType}): {$Value} ({$ValueType})",
-                    Property.Name, 
-                    _valueType, 
-                    value,
-                    value?.GetType());
-            }
+            }, (value, priority), val => !object.Equals(val.value, _value));
```

Comment:
As far as I can see the `priority` part of `TValue` isn't actually being used... What am I missing?

Reply:
Looks like I accidentally captured the `priority` parameter instead of using the priority from the passed in update. It's getting passed in so we can preserve the priority of each scheduled set.